src/SFML/Network/Ftp.cpp

```diff
@@ -115,10 +117,24 @@ const char* sfFtpDirectoryResponse_getMessage(const sfFtpDirectoryResponse* ftpD
 
 ////////////////////////////////////////////////////////////
 const char* sfFtpDirectoryResponse_getDirectory(const sfFtpDirectoryResponse* ftpDirectoryResponse)
+{
+    CSFML_CHECK_RETURN(ftpDirectoryResponse, nullptr);
+    return strdup(ftpDirectoryResponse->This.getDirectory().string().c_str());
+}
+
+
+////////////////////////////////////////////////////////////
+const sfChar32* sfFtpDirectoryResponse_getDirectoryUnicode(const sfFtpDirectoryResponse* ftpDirectoryResponse)
 {
     CSFML_CHECK_RETURN(ftpDirectoryResponse, nullptr);
 
-    return ftpDirectoryResponse->This.getDirectory().c_str();
+    std::u32string str = sf::String(ftpDirectoryResponse->This.getDirectory().c_str()).toUtf32();
+
+    std::size_t size = sizeof(sfChar32) * str.length();
+    sfChar32* ret = static_cast<sfChar32*>(malloc(size));
+    memcpy(ret, str.c_str(), size);
```

Comment:
Could this use the new conversion functions you wrote?

Reply:
Done.